contracts/external/lib/TypedSignature.sol

```diff
@@ -0,0 +1,117 @@
+/*
+
+    Copyright 2019 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.7;
+pragma experimental ABIEncoderV2;
+
+import { Require } from "../../protocol/lib/Require.sol";
+
+
+/**
+ * @title TypedSignature
+ * @author dYdX
+ *
+ * Library to unparse typed signatures
+ */
+library TypedSignature {
+
+    // ============ Constants ============
+
+    bytes32 constant private FILE = "TypedSignature";
+
+    // prepended message with the length of the signed hash in decimal
+    bytes constant private PREPEND_DEC = "\x19Ethereum Signed Message:\n32";
+
+    // prepended message with the length of the signed hash in hexadecimal
+    bytes constant private PREPEND_HEX = "\x19Ethereum Signed Message:\n\x20";
+
+    // Number of bytes in a typed signature
+    uint256 constant private NUM_SIGNATURE_BYTES = 66;
+
+    // ============ Enums ============
+
+    enum SignatureType {
+        Invalid,
+        NoPrepend,
+        Decimal,
+        Hexadecimal,
+        Unsupported
```

Comment:
what are `Invalid` and `Unsupported` for ...?

Reply:
Guess it's not really necessary but it's similar to what we had in protocol_v1 and what 0x has

Reply:
Removed